src/main/scala/is/hail/rvd/RVDContext.scala

```diff
@@ -1,20 +1,58 @@
 package is.hail.rvd
 
-import is.hail.annotations.Region
+import is.hail.annotations.{Region, RegionValueBuilder}
+import is.hail.sparkextras.ResettableContext
+
+import scala.collection.mutable
 
 object RVDContext {
   def default: RVDContext = new RVDContext(Region())
 
   def fromRegion(region: Region): RVDContext = new RVDContext(region)
 }
 
-// NB: must be *Auto*Closeable because calling close twice is undefined behavior
-// (see AutoCloseable javadoc)
-class RVDContext(r: Region) extends AutoCloseable {
+class RVDContext(r: Region) extends ResettableContext {
+  private[this] val children: mutable.ArrayBuffer[AutoCloseable] = new mutable.ArrayBuffer()
+
+  private[this] def own(child: AutoCloseable): Unit = children += child
+
+  own(r)
+
+  def freshContext: RVDContext = {
+    val ctx = RVDContext.default
+    own(ctx)
+    ctx
+  }
+
   def region: Region = r // lifetime: element
 
   def partitionRegion: Region = r // lifetime: partition
```

Comment:
I've been wondering about this.  Does it get used?  How?

Reply:
I'll remove it. I was thinking it would be useful but it's not correct here (the region gets cleared too soon to be partition-lived). I thought it might be useful for methods that need a little scratch space per-partition (like `localKeySort` that holds on to values between calls to `next`)